src/Functions/FunctionStringCompare.cpp

```diff
@@ -0,0 +1,414 @@
+        }
+
+        col_result->getData().resize(arguments[0].column->size());
+        auto & result_array = col_result->getData();
+
+        if (col_str1 && col_str2_const)
+            executeStringConst<false>(*col_str1, str1_offset, String(col_str2_const->getDataAt(0)), str2_offset, num_bytes, result_array);
+        else if (col_str1_const && col_str2)
+            executeStringConst<true>(*col_str2, str2_offset, String(col_str1_const->getDataAt(0)), str1_offset, num_bytes, result_array);
+        else if (col_str1 && col_str2)
+            executeStringString(*col_str1, str1_offset, *col_str2, str2_offset, num_bytes, result_array);
+        else if (fixed_str1_col && fixed_str2_col)
+            executeFixedStringFixedString(*fixed_str1_col, str1_offset, *fixed_str2_col, str2_offset, num_bytes, result_array);
+        else if (fixed_str1_col && col_str2)
+            executeFixedStringString<false>(*fixed_str1_col, str1_offset, *col_str2, str2_offset, num_bytes, result_array);
+        else if (col_str1 && fixed_str2_col)
+            executeFixedStringString<true>(*fixed_str2_col, str2_offset, *col_str1, str1_offset, num_bytes, result_array);
+        else if (fixed_str1_col && col_str2_const)
+            executeFixedStringConst<false>(*fixed_str1_col, str1_offset, String(col_str2_const->getDataAt(0)), str2_offset, num_bytes, result_array);
+        else if (fixed_str2_col && col_str1_const)
+            executeFixedStringConst<true>(*fixed_str2_col, str2_offset, String(col_str1_const->getDataAt(0)), str1_offset, num_bytes, result_array);
+        else if (col_str1_const && col_str2_const)
+            executeConstConst(
+                String(col_str1_const->getDataAt(0)),
+                str1_offset,
+                String(col_str2_const->getDataAt(0)),
+                str2_offset,
+                num_bytes,
+                arguments[0].column->size(),
+                result_array);
+        else
+            throw Exception(
+                ErrorCodes::ILLEGAL_COLUMN,
+                "Illegal columns {} and {} of arguments of function {}",
+                arguments[0].column->getName(),
+                arguments[1].column->getName(),
+                getName());
+        return std::move(col_result);
+    }
+
+private:
+    static bool isConstColumn(const IColumn & column) { return column.isConst(); }
+
+    template <bool reverse>
+    static Int8 normalComparison(const char * str1, size_t str1_length, const char * str2, size_t str2_length)
+    {
+        if constexpr (reverse)
+            return static_cast<Int8>(memcmpSmallLikeZeroPaddedAllowOverflow15(str2, str2_length, str1, str1_length));
+        else
+            return static_cast<Int8>(memcmpSmallLikeZeroPaddedAllowOverflow15(str1, str1_length, str2, str2_length));
+    }
+
+    template <bool reverse>
+    static bool isOverflowComparison(size_t str1_length, size_t str1_offset, size_t str2_length, size_t str2_offset, Int8 & result)
```

Comment:
Please help future readers and add a comment what the return value of function `isOverflowComparison` means.